### PR TITLE
feat: 字段标记(filed)可使用正则进行匹配

### PR DESCRIPTION
**Pull request by stone-lyl**
### 👀 PR includes

✨ Feature

- [X] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [ ] Test case
- [X] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description
feat: 新增功能字段标记可使用正则进行匹配。
feat: 一个字段 ID 多次匹配到同一范围的字段标记规则，以最后一个规则为准
### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |
|image|image|
|image|image|

### 🔍 Self Check before Merge

- [ ] Add or update relevant Docs.
- [ ] Add or update relevant Demos.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: s2-site/docs/manual/basic/conditions.zh.md [hunk 1]
```diff
@@ -57,8 +57,8 @@ const s2options = {
 
 `field` 用于指定将字段标记应用于哪些字段上，其取值范围会因表的形态不同而不同：
 
-* 对于透视表，`field` 取值范围是 `values`
-* 对于明细表，`field` 取值范围是 `columns`
+* 对于透视表，`field` 取值范围或正则匹配范围是 `values`
+* 对于明细表，`field` 取值范围或正则匹配范围是 `columns`
```
Comment (wjgogogo) [inline s2-site/docs/manual/basic/conditions.zh.md:61]: 文档里面说明一下，优先级是越后面越高

**Review by xingwanying — DISMISSED** [review]

File: s2-site/docs/manual/basic/conditions.zh.md (hunk 1 above)
Comment (stone-lyl) [reply s2-site/docs/manual/basic/conditions.zh.md:61]: 我觉得你的文档写的好规整，我感觉那里都不好插入这样一句话🐶

Comment (stone-lyl) [reply s2-site/docs/manual/basic/conditions.zh.md:62]: 我还是强行加入了

**Review by lijinke666 — APPROVED** [review]

**Comment by lijinke666** [comment]
@stone-lyl 后面补点测试

**Pull request merged**